### ISSUE TITLE
NPE crash

### ISSUE THREAD

**Issue by NiaNingXue** [AppLovin/AppLovin-MAX-SDK-Android#194]
implementation 'com.applovin:applovin-sdk:11.1.2'

```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Set com.applovin.impl.a.a.a(com.applovin.impl.a.a$c, java.lang.String)' on a null object reference

com.applovin.impl.adview.activity.b.c.a
com.applovin.impl.adview.activity.b.c.a
com.applovin.impl.adview.activity.b.c.h
com.applovin.impl.adview.activity.b.a$4$1.run
com.applovin.impl.sdk.e.z.run
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1167
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:641
java.lang.Thread.run Thread.java:764
```

**Comment by thomasmso** [comment]
Hi @NiaNingXue - We're investigating. In the meantime, our latest SDK is v11.1.3 - do you mind updating to that?

**Comment by NiaNingXue** [comment]
@thomasmso I have so many NPE crash, so it is better to update when fix this bug

**Comment by sijifairy** [comment]
Many same NPE either, please fix asap

**Comment by thomasmso** [comment]
@sijifairy - What SDK version are you on? And what % of users is this crash occurring on? @NiaNingXue

**Comment by sijifairy** [comment]
@thomasmso 11.1.2  3% device crashed

Today I release a new version using AppLovin 11.1.3, there's a similar NPE crashed one time till now:

java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3839)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3997)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:85)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2380)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:257)
  at android.app.ActivityThread.main (ActivityThread.java:8335)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:626)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1032)
Caused by: java.lang.NullPointerException: 
  at com.applovin.impl.adview.activity.b.e.C (Unknown Source:2)
  at com.applovin.impl.adview.activity.b.e.l (Unknown Source)
  at com.applovin.impl.adview.activity.b.e.h (Unknown Source:11)
  at com.applovin.impl.adview.activity.b.a$4$1.run (Unknown Source:11)
  at com.applovin.sdk.AppLovinSdkUtils.runOnUiThread (Unknown Source:8)
  at com.applovin.sdk.AppLovinSdkUtils.runOnUiThread (Unknown Source:1)
  at com.applovin.impl.adview.activity.b.a$4.onActivityCreated (Unknown Source:39)
  at com.applovin.impl.sdk.a.onActivityCreated (Unknown Source:23)
  at android.app.Application.dispatchActivityCreated (Application.java:390)
  at android.app.Activity.dispatchActivityCreated (Activity.java:1414)
  at android.app.Activity.onCreate (Activity.java:1712)
  at androidx.core.app.ComponentActivity.onCreate (ComponentActivity.java:85)
  at androidx.activity.ComponentActivity.onCreate (ComponentActivity.java:120)
  at androidx.fragment.app.FragmentActivity.onCreate (FragmentActivity.java:275)
  at androidx.appcompat.app.AppCompatActivity.onCreate (AppCompatActivity.java:85)

**Comment by adxcorp** [comment]
I also using 11.1.3 version. I got same Exception from AppLovin SDK.

Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Set com.applovin.impl.a.a.a(com.applovin.impl.a.a$c, java.lang.String)' on a null object reference
       at com.applovin.impl.adview.activity.b.c.a(:2)
       at com.applovin.impl.adview.activity.b.c.a(:2)
       at com.applovin.impl.adview.activity.b.c.h(:4)
       at com.applovin.impl.adview.activity.b.a$4$1.run(:11)
       at com.applovin.sdk.AppLovinSdkUtils.runOnUiThread(:8)
       at com.applovin.sdk.AppLovinSdkUtils.runOnUiThread(:1)
       at com.applovin.impl.adview.activity.b.a$4.onActivityCreated(:39)
       at com.applovin.impl.sdk.a.onActivityCreated(:23)
       at android.app.Application.dispatchActivityCreated(Application.java:245)
       at android.app.Activity.onCreate(Activity.java:1108)
       at androidx.core.app.ComponentActivity.onCreate(ComponentActivity.java:85)
       at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:323)
       at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:273)
       at com.malangstudio.alarmmon.BaseActivity.onCreate(BaseActivity.java:24)
       at com.malangstudio.alarmmon.SplashActivity.onCreate(SplashActivity.java:20)
       at android.app.Activity.performCreate(Activity.java:7327)
       at android.app.Activity.performCreate(Activity.java:7318)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3096)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3259)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1950)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7073)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)

**Comment by santoshbagadi** [comment]
Hi @adxcorp , we are actively investigating this issue and will have a fix in the next release. Curious if you are able to reproduce the issue locally. 

Also, would you be able to share your app's `AndroidManifest.xml` file?

**Comment by htzz271** [comment]
11.1.3 version crash

Caused by java.lang.NullPointerException
Attempt to invoke virtual method 'java.util.Set com.applovin.impl.a.a.a(com.applovin.impl.a.a$c, java.lang.String)' on a null object reference
com.applovin.impl.adview.activity.b.c.a (c.java:2)
com.applovin.impl.adview.activity.b.c.a (c.java:2)
com.applovin.impl.adview.activity.b.c.h (c.java:4)
com.applovin.impl.adview.activity.b.a$4$1.run (a.java:11)
com.applovin.sdk.AppLovinSdkUtils.runOnUiThread (AppLovinSdkUtils.java:8)
com.applovin.sdk.AppLovinSdkUtils.runOnUiThread (AppLovinSdkUtils.java:1)
com.applovin.impl.adview.activity.b.a$4.onActivityCreated (a.java:39)
com.applovin.impl.sdk.a.onActivityCreated (a.java:23)
android.app.Application.dispatchActivityCreated (Application.java:392)

**Comment by santoshbagadi** [comment]
Hi @adxcorp , would you mind trying to reproduce the issue as shown below

1. Show an AppLovin fullscreen ad.
2. Go to the home screen while the ad is playing.
3. Launch the app again from the launcher by clicking the app icon (instead of going to the app from recents) and see if that causes the crash.

The reason for requesting the AndroidManifest is to check it it is due to a `android:launchMode` flag that is causing the issue.

**Comment by adxcorp** [comment]
Hi @santoshbagadi 

        <activity
            android:name="com.malangstudio.alarmmon.AlarmMonActivity"
            android:configChanges="keyboard|keyboardHidden|screenSize"
            android:hardwareAccelerated="true"
            android:launchMode="singleTask"
            android:screenOrientation="user" />

        <activity
            android:name=".cocos2djs.AlarmAlertActivity"
            android:autoRemoveFromRecents="true"
            android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize"
            android:directBootAware="true"
            android:excludeFromRecents="true"
            android:launchMode="singleInstance"
            android:screenOrientation="portrait" />

I guess singleTask is occur this issue. Is right?

**Comment by adxcorp** [comment]
I found this code at our application, I guess it also occur same issue.

startActivity(Intent.makeRestartActivityTask(new ComponentName(this, SplashActivity.class)));

**Comment by santoshbagadi** [comment]
Thank you for all the info @adxcorp , we have a similar setup but we have not been able to reproduce the issue. Were you able to reproduce the issue by trying the steps mentioned above?

**Comment by adxcorp** [comment]
@santoshbagadi We also can't reproduce that. But our user is affected this issue :(

**Comment by santoshbagadi** [comment]
@adxcorp , @htzz271 , @NiaNingXue , @sijifairy  we have just released version [11.2.0](https://github.com/AppLovin/AppLovin-MAX-SDK-Android/releases/tag/release_11_2_0) of our SDK that fixes this issue.
